src/libexpr/nixexpr.cc

```diff
@@ -328,31 +358,56 @@ void ExprOpHasAttr::bindVars(EvalState & es, const std::shared_ptr<const StaticE
             i.expr->bindVars(es, env);
 }
 
+std::shared_ptr<const StaticEnv> ExprAttrs::bindInheritSources(
+    EvalState & es, const std::shared_ptr<const StaticEnv> & env)
+{
+    if (!inheritFromExprs)
+        return nullptr;
+
```

Comment:
This seems like helpful context:

```suggestion

    // The inherit (from) values are inserted into an Env of its own, which
    // does not introduce any variable names.
```